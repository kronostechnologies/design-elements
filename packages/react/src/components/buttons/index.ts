export { Size } from './abstract/types';
export { ButtonProps, ButtonType, Type } from './types';
export { getBaseButtonStyles } from './abstract/styled';
export { getButtonTypeStyles, getButtonStyles } from './styled';
export { AbstractButton } from './abstract/abstract-button';
export { Button } from './button';
export { SearchButton } from './search-button';
export { IconButton } from './icon-button';
