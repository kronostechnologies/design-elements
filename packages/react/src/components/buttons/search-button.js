"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var abstract_button_1 = require("./abstract-button");
var StyledButton = styled_components_1.default(abstract_button_1.AbstractButton)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    background: rgb(255, 255, 255);\n    border-color: rgb(217, 221, 226);\n    color: rgb(99, 114, 130);\n\n    &:focus,\n    &:hover {\n        background-color: rgb(217, 221, 226);\n        border-color: rgb(217, 221, 226);\n        color: rgb(99, 114, 130);\n    }\n\n    &:disabled {\n        &,\n        &:focus,\n        &:hover {\n            background-color: rgb(242, 243, 249);\n            border-color: rgb(217, 221, 226);\n            color: rgb(156, 167, 180);\n        }\n    }\n"], ["\n    background: rgb(255, 255, 255);\n    border-color: rgb(217, 221, 226);\n    color: rgb(99, 114, 130);\n\n    &:focus,\n    &:hover {\n        background-color: rgb(217, 221, 226);\n        border-color: rgb(217, 221, 226);\n        color: rgb(99, 114, 130);\n    }\n\n    &:disabled {\n        &,\n        &:focus,\n        &:hover {\n            background-color: rgb(242, 243, 249);\n            border-color: rgb(217, 221, 226);\n            color: rgb(156, 167, 180);\n        }\n    }\n"])));
var SearchButton = function (_a) {
    var children = _a.children, className = _a.className, disabled = _a.disabled, onClick = _a.onClick;
    var handleClick = function () { onClick && onClick(); };
    return (<StyledButton className={className} disabled={disabled} onClick={handleClick}>
            {children}
        </StyledButton>);
};
exports.SearchButton = SearchButton;
var templateObject_1;
