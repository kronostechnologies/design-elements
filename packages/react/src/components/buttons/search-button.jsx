import React from 'react';
import styled from 'styled-components';

import { AbstractButton } from './abstract-button';

const Button = styled(AbstractButton)`
  background: rgb(255, 255, 255);
  border-color: rgb(217, 221, 226);
  color: rgb(99, 114, 130);

  &:focus,
  &:hover {
    background-color: rgb(217, 221, 226);
    border-color: rgb(217, 221, 226);
    color: rgb(99, 114, 130);
  }

  &:disabled {
    &,
    &:focus,
    &:hover {
      background-color: rgb(242, 243, 249);
      border-color: rgb(217, 221, 226);
      color: rgb(156, 167, 180);
    }
  }
`;

const SearchButton = ({ children, disabled, onClick, ...props }) => (
    <Button {...props} disabled={disabled} onClick={onClick}>
        {children}
    </Button>
);

export { SearchButton };
