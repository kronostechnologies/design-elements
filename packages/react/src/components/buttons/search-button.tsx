import React, { ReactNode } from 'react';

import styled from 'styled-components';
import {Â AbstractButton } from './abstract-button';

interface ButtonProps {
    label?: string;
    children?: ReactNode;
    className: string;
    disabled?: boolean;

    onClick?(): void;
}

const StyledButton = styled(AbstractButton)`
    ${props => `
    background: ${props.theme.greys.white};
    border-color: ${props.theme.greys.grey};
    color: ${props.theme.greys['mid-grey']};

    &:disabled {
        &,
        &:focus,
        &:hover {
            background-color: ${props.theme.greys.grey};
            border-color: ${props.theme.greys.grey};
            color: ${props.theme.greys['mid-grey']};
        }

        &:disabled {
            &,
            &:focus,
            &:hover {
                background-color: ${props.theme.greys['light-grey']};
                border-color: ${props.theme.greys.grey};
                color: ${props.theme.greys['mid-grey']};
            }
        }
    }`}
`;

const SearchButton = ({ children, className, disabled, label, onClick }: ButtonProps) => {
    const handleClick = () => { onClick && onClick(); };

    return (
        <StyledButton className={className} disabled={disabled} onClick={handleClick}>
            {children}{label}
        </StyledButton>
    );
};

export { SearchButton };
