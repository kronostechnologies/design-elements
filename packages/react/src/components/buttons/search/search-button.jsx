import React from 'react';
import styled from 'styled-components';

import Abstract from '../abstract';

const SearchButton = styled(Abstract)`
  background: rgb(255, 255, 255);
  border-color: rgb(217, 221, 226);
  color: rgb(99, 114, 130);

  &:focus,
  &:hover {
    background-color: rgb(217, 221, 226);
    border-color: rgb(217, 221, 226);
    color: rgb(99, 114, 130);
  }

  &:disabled {
    &,
    &:focus,
    &:hover {
      background-color: rgb(242, 243, 249);
      border-color: rgb(217, 221, 226);
      color: rgb(156, 167, 180);
    }
  }
`;

export default ({ children, disabled, onClick, ...props }) => (
    <SearchButton {...props} disabled={disabled} onClick={onClick}>
        {children}
    </SearchButton>
);
