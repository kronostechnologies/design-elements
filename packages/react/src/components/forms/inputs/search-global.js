"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var search_input_1 = require("./search-input");
var SearchGlobal = function (_a) {
    var disabled = _a.disabled, id = _a.id, label = _a.label, onSearch = _a.onSearch;
    return (<search_input_1.SearchInput disabled={disabled} hasButton id={id} label={label} onSearch={onSearch}/>);
};
exports.SearchGlobal = SearchGlobal;
