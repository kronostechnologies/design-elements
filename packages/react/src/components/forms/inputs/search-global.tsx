import React from 'react';

import { SearchInput, SearchInputProps } from './search-input';
const uuidv1 = require('uuid/v1');

const SearchGlobal = ({ disabled, label, searchCallback }: SearchInputProps) => {
    const id = uuidv1();
    return (
        <SearchInput
            disabled={disabled}
            hasButton
            id={id}
            label={label}
            searchCallback={searchCallback}
        />
    );
};

export { SearchGlobal };
