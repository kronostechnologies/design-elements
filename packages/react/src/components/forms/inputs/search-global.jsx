import React from 'react';
import styled from 'styled-components';

import { SearchInput } from './search-input';
import { SearchButton } from '../../buttons/search-button.jsx';

const SearchSubmit = styled(SearchButton)`
  border-left: 0;
  border-radius: 0 0.25rem 0.25rem 0;
  position: relative;
`;

const SearchGlobal = ({ disabled, id, label, onSubmit }) => (
    <SearchInput
        disabled={disabled}
        id={id}
        label={label}
        onSubmit={onSubmit}
        global
    >
        <SearchSubmit
            disabled={disabled}
            type="submit"
            onClick={onSubmit}
        >
            Search
        </SearchSubmit>
    </SearchInput>
);

export { SearchGlobal };
