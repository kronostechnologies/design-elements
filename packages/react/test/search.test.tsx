import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { SearchButton } from '../src/components/buttons/search-button';
import { SearchGlobal } from '../src/components/forms/inputs/search-global';

describe('SearchGlobal', () => {
    test('Search callback is called when search button is clicked', () => {
        const mockCallback = jest.fn().mockImplementation((value: string) => `Searching for: ${value}`);
        const wrapper = mount(
            <SearchGlobal initialValue="foo" label="Search" onSearch={mockCallback} />,
        );
        const button = wrapper.find(SearchButton);

        button.simulate('click');

        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveReturnedWith('Searching for: foo');
    });

    test('Search callback is called when Enter is pressed in input', () => {
        const mockCallback = jest.fn().mockImplementation((value: string) => `Searching for: ${value}`);
        const wrapper = mount(
            <SearchGlobal initialValue="bing" label="Search" onSearch={mockCallback} />,
        );
        const input = wrapper.find('input');

        input.simulate('keyDown', { keyCode: 13 });

        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveReturnedWith('Searching for: bing');
    });

    test('Contains initial value', () => {
        const wrapper = mount(
            <SearchGlobal initialValue="foo" label="Search" onSearch={() => {}} />,
        );
        const input = wrapper.find('input');

        expect(input.props().value).toBe('foo');
    });

    test('Reset buttons clears input value', () => {
        const wrapper = mount(
            <SearchGlobal initialValue="foo" label="Search" onSearch={() => {}} />,
        );
        const reset = wrapper.find('[data-testid="resetButton"]').at(1);
        reset.simulate('click');

        expect(wrapper.contains('foo')).toBe(false);
    });

    test('Matches the snapshot', () => {
        const tree = renderer.create(
            <SearchGlobal initialValue="foo" label="Search" onSearch={() => {}} />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
