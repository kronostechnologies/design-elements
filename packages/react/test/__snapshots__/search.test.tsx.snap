// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchGlobal Matches the snapshot 1`] = `
<div
  className="sc-ifAKCX jojDhl"
>
  <div
    className="sc-EHOje eNdeiV"
  >
    <label
      className="sc-bxivhb ehRUut"
      htmlFor="searchbar_global"
    >
      <svg
        className="sc-bZQynM gQLQiz"
      />
      <span
        aria-hidden="false"
        className="sc-htpNat gPwbuH"
      >
        Search
      </span>
    </label>
    <input
      autoComplete="on"
      className="sc-htoDjs hECHJA"
      id="searchbar_global"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="search"
      value="foo"
    />
    <button
      className="sc-dnqmqq eDuaOl"
      data-testid="resetButton"
      onClick={[Function]}
    >
      <svg
        className="sc-gzVnrw cODjhC"
      />
      <span
        aria-hidden="false"
        className="sc-htpNat gPwbuH"
      >
        Reset
      </span>
    </button>
  </div>
  <button
    className="sc-bdVaJa primary sc-iwsKbI htioOs sc-bwzfXH xbjaQ"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
